exporter/opentelemetry-exporter-zipkin/src/opentelemetry/exporter/zipkin/__init__.py

```diff
@@ -60,384 +64,101 @@
-        transport_format: Union[
-            TRANSPORT_FORMAT_JSON, TRANSPORT_FORMAT_PROTOBUF, None
-        ] = None,
+        service_name: Optional[str] = None,
+        endpoint: Optional[str] = None,
+        encoding: Optional[Encoding] = None,
+        local_node_ipv4: IpInput = None,
+        local_node_ipv6: IpInput = None,
+        local_node_port: Optional[int] = None,
+        max_tag_value_length: Optional[int] = None,
     ):
-        self.service_name = service_name
-        if url is None:
-            self.url = Configuration().EXPORTER_ZIPKIN_ENDPOINT or DEFAULT_URL
-        else:
-            self.url = url
+        if service_name is None:
+            service_name = (
+                Configuration().EXPORTER_ZIPKIN_SERVICE_NAME
+                or DEFAULT_SERVICE_NAME
+            )
+        self.local_node = NodeEndpoint(
+            service_name, local_node_ipv4, local_node_ipv6, local_node_port
+        )
 
-        self.port = urlparse(self.url).port
+        if endpoint is None:
+            endpoint = (
+                Configuration().EXPORTER_ZIPKIN_ENDPOINT or DEFAULT_ENDPOINT
+            )
+        self.endpoint = endpoint
 
-        self.ipv4 = ipv4
-        self.ipv6 = ipv6
-        self.retry = retry
-        self.max_tag_value_length = max_tag_value_length
+        if encoding is None:
+            env_encoding = Configuration().EXPORTER_ZIPKIN_ENCODING
+            if env_encoding is not None:
+                encoding = Encoding(env_encoding)
+            else:
+                encoding = DEFAULT_ENCODING
 
-        if transport_format is None:
-            self.transport_format = (
-                Configuration().EXPORTER_ZIPKIN_TRANSPORT_FORMAT
-                or TRANSPORT_FORMAT_JSON
-            )
-        else:
-            self.transport_format = transport_format
+        if encoding == Encoding.V1_JSON:
+            self.encoder = JsonV1Encoder(max_tag_value_length)
+        elif encoding == Encoding.V2_JSON:
+            self.encoder = JsonV2Encoder(max_tag_value_length)
+        elif encoding == Encoding.V2_PROTOBUF:
+            self.encoder = ProtobufEncoder(max_tag_value_length)
 
     def export(self, spans: Sequence[Span]) -> SpanExportResult:
-        if self.transport_format == TRANSPORT_FORMAT_JSON:
-            content_type = "application/json"
-        elif self.transport_format == TRANSPORT_FORMAT_PROTOBUF:
-            content_type = "application/x-protobuf"
-        else:
-            logger.error("Invalid transport format %s", self.transport_format)
-            return SpanExportResult.FAILURE
-
         result = requests.post(
-            url=self.url,
-            data=self._translate_to_transport_format(spans),
-            headers={"Content-Type": content_type},
+            url=self.endpoint,
+            data=self.encoder.serialize(spans, self.local_node),
+            headers={"Content-Type": self.encoder.content_type()},
         )
```

Comment:
Out of scope for this PR. Having `keep-alive` improves any performance?

Reply:
Potentially in a batch export config. Would likely tweak to create and manage a `sessions.Session()` in the exporter's `init()` and `shutdown()` methods.